app/assets/javascripts/calendars/guiders-appointments.es6

```diff
@@ -34,19 +39,80 @@
             color: 'green',
             rendering: 'background'
           }
-        ]
+        ],
+        resources: (...args) => this.resources(...args)
       };
 
       super.start(el);
 
       this.eventChanges = [];
+      this.filterList = [];
       this.actionPanel = $('[data-action-panel]');
       this.saveWarningMessage = 'You have unsaved changes - Save, or undo the changes.';
+      this.filterButton = $('.fc-filter-button');
+      this.filterPanel = $('.resource-calendar-filter');
 
+      this.bindEvents();
       this.setCalendarToCorrectHeight();
       this.setupUndo();
     }
 
+    resources(callback) {
+      $.get('/guiders', this.handleResources.bind(this, callback));
+    }
+
+    handleResources(callback, resources) {
+      const filteredResources = resources.filter(this.filterResources.bind(this));
+
+      if (filteredResources.length === 0) {
+        return callback(resources);
+      }
+
+      callback(filteredResources);
+    }
+
+    filterResources(resource) {
+      return $.inArray(resource.id, $.map(this.filterList, (id) => { return parseInt(id); })) > -1;
```

Comment:
Bit hard to understand this, could you extract a variable for the map stuff?

`
let ids = $.map(this.filterList, (id) => { return parseInt(id); });
return $.inArray(resource.id, ids) > -1;
`

Or better yet, store `ints` in `filterList`? I feel like I'm missing something entirely here, or is that logical ^^